resolve/LinSolverDirectRocSparseILU0.cpp

```diff
@@ -0,0 +1,268 @@
+
+    // Create matrix descriptor for U
+    rocsparse_create_mat_descr(&descr_U_);
+    rocsparse_set_mat_fill_mode(descr_U_, rocsparse_fill_mode_upper);
+    rocsparse_set_mat_diag_type(descr_U_, rocsparse_diag_type_non_unit);
+
+    // Create matrix info structure
+    rocsparse_create_mat_info(&info_A_);
+
+    size_t buffer_size_A;
+    size_t buffer_size_L;
+    size_t buffer_size_U;
+
+    status_rocsparse_ = rocsparse_dcsrilu0_buffer_size(workspace_->getRocsparseHandle(), 
+                                                       n, 
+                                                       nnz, 
+                                                       descr_A_,
+                                                       d_ILU_vals_, //vals_, 
+                                                       A_->getRowData(ReSolve::memory::DEVICE), 
+                                                       A_->getColData(ReSolve::memory::DEVICE), 
+                                                       info_A_, 
+                                                       &buffer_size_A);
+
+    error_sum += status_rocsparse_;
+    status_rocsparse_ = rocsparse_dcsrsv_buffer_size(workspace_->getRocsparseHandle(), 
+                                                     rocsparse_operation_none, 
+                                                     n, 
+                                                     nnz, 
+                                                     descr_L_,
+                                                     d_ILU_vals_, //vals_, 
+                                                     A_->getRowData(ReSolve::memory::DEVICE), 
+                                                     A_->getColData(ReSolve::memory::DEVICE), 
+                                                     info_A_, 
+                                                     &buffer_size_L);
+    error_sum += status_rocsparse_;
+
+    status_rocsparse_ = rocsparse_dcsrsv_buffer_size(workspace_->getRocsparseHandle(), 
+                                                     rocsparse_operation_none, 
+                                                     n, 
+                                                     nnz, 
+                                                     descr_U_,
+                                                     d_ILU_vals_, //vals_, 
+                                                     A_->getRowData(ReSolve::memory::DEVICE), 
+                                                     A_->getColData(ReSolve::memory::DEVICE), 
+                                                     info_A_,
+                                                     &buffer_size_U);
+    error_sum += status_rocsparse_;
+
+    size_t buffer_size = static_cast<size_t>(max(static_cast<int>(buffer_size_A),static_cast<int>(max(static_cast<int>(buffer_size_L),static_cast<int>(buffer_size_U)))));
+    mem_.allocateBufferOnDevice(&buffer_, buffer_size);
```

Comment:
I believe this could be much nicer fix for the conversion warnings:
```suggestion
    size_t buffer_size = std::max(buffer_size_A, std::max(buffer_size_L, buffer_size_U));
```

Reply:
CC @cameronrutherford